src/System.Private.ServiceModel/tests/Common/src/TestTypes.cs

```diff
@@ -688,3 +688,20 @@ public string StringValue
         set { _stringValue = value; }
     }
 }
+
+public class DuplexTaskReturnServiceCallback : IWcfDuplexTaskReturnCallback
+{
+    public Task<Guid> ServicePingCallback(Guid guid)
+    {
+        // This returns the guid to the service which called this callback.
+        // We could return Task.FromResult(guid) but that means we could execute the 
+        // completion on the same thread. But if someone is using a task it means they 
+        // would potentially have the completion on another thread.
+        return Task.Run<Guid>(() => guid);
+    }
+
+    public Task<Guid> ServicePingFaultCallback(Guid guid)
+    {
+        throw new FaultException<FaultDetail>(new FaultDetail("Throwing a Fault Exception from the Callback method."), "Reason: Testing FaultException returned from Duplex Callback");
```

Comment:
From my testing and fixing of bugs based on this new test, we need another slightly different variation. This callback throws as soon as it's called which means it doesn't return a faulted Task. A second variation where it uses `Task.FromException` to return a faulted Task would be beneficial as they behave differently in the product.